### ISSUE TITLE
Support for IAM Policy for Lambda Authorizer (v1)

### ISSUE THREAD

**Issue by missourian55** [aws/aws-lambda-java-libs#184]
Appreciate you guys added few event types to support the LambdaAuthorizer, Any plans to add a event type for IAM policy response which will be compatible with REST API?

Currently I am using this class from blueprints 
https://github.com/awslabs/aws-apigateway-lambda-authorizer-blueprints/blob/master/blueprints/java/src/io/AuthPolicy.java

```
APIGatewayCustomAuthorizerEvent (Rest) 
APIGatewayV2CustomAuthorizerEvent (http)
SimpleIAMPolicyResponse (http)
IAMPolicyResponse (rest) <---- missing
```

**Comment by msailes** [comment]
Hi missourian55,

Yes, there is an open PR for this response.

Mark

**Comment by missourian55** [comment]
Any ETA on when this will be merged and released? We love to have this soon to remove some custom implementation in our stack.

**Comment by msailes** [comment]
Hi missourian55,

For a couple of reason this is proving a little annoying. In the short term this class should meet your needs.

https://github.com/awslabs/aws-apigateway-lambda-authorizer-blueprints/blob/master/blueprints/java/src/io/AuthPolicy.java

Please reply back if you have any questions.

**Comment by msailes** [comment]
This has been released in version 3.8.0

**Comment by missourian55** [comment]
Just gave `IamPolicyResponse` a try on **Rest API**. Getting below error

` Execution failed due to configuration error: Unable to parse given policy: Syntax error at position (-1,-1)`


```properties
{"principalId":"some","policyDocument":{"Version":"2012-10-17","Statement":[{"Condition":null,"Action":"execute-api:Invoke","Resource":["arn:aws:execute-api:us-east-1:123456789012:hr2mbvlnw6/some/POST/somemethod"],"Effect":"Allow"}]},"context":null}
```

**Comment by msailes** [comment]
My apologies, the response required for the REST API has an additional field `"usageIdentifierKey": "{api-key}"` and is a required field.

I'll create a new PR.

**Comment by missourian55** [comment]
@msailes Thanks for looking into this. Currently we are using [AuthPolicy](https://github.com/awslabs/aws-apigateway-lambda-authorizer-blueprints/blob/master/blueprints/java/src/io/AuthPolicy.java) for REST api and we never populated `usageIdentifierKey`. Is this a new breaking change? 

If I need to populate this where I should get this value from?

**Comment by msailes** [comment]
Do you have an example handler that you can share which exhibits this error?

**Comment by missourian55** [comment]
is it possible to share in any private mode? Otherwise I need to do lot of scrubbing to eliminate my company info

**Comment by msailes** [comment]
You can contact me on my username @amazon.co.uk

**Comment by missourian55** [comment]
Done

**Comment by wilerson** [comment]
Any ETA on merging and releasing #233 ?

**Comment by msailes** [comment]
@wilerson we hope to release this in June.

**Comment by msailes** [comment]
this has been released in v3.9.0
